File: gpt/new_tokens.py
token_memory = {}

def generate_token(word):
    return sum(ord(c) for c in word)

def learn_word(word):
    token = generate_token(word)
    token_memory[word] = token
    with open("memory.txt", "a") as f:
        f.write(f"{word}:{token}\n")
    print(f"Aprendi: {word} -> Token: {token}")

print("LumaGPT (Python) pronto! Digite palavras:")

while True:
    inp = input("> ")
    if inp.lower() == "sair":
        break
    learn_word(inp)

File: gpt/new_tokens.jl
token_memory = Dict{String, Int}()

function generate_token(word::String)
    sum(Int(c) for c in word)
end

function learn_word(word::String)
    token = generate_token(word)
    token_memory[word] = token
    open("memory.txt", "a") do f
        write(f, "$word:$token\n")
    end
    println("Aprendi: $word -> Token: $token")
end

println("LumaGPT (Julia) pronto! Digite palavras:")

while true
    print("> ")
    input = readline()
    if lowercase(input) == "sair"
        break
    end
    learn_word(input)
end
